api/src/main/java/org/openmrs/module/ModuleUtil.java

```diff
@@ -571,7 +577,15 @@ public static void expandJar(File fileToExpand, File tmpModuleDir, String name,
 					String entryName = jarEntry.getName();
 					// trim out the name path from the name of the new file
 					if (!keepFullPath && name != null) {
-						entryName = entryName.replaceFirst(name, "");
+						if (name.equals(entryName)) {
+							int lastSlash = name.lastIndexOf('/');
+							if (lastSlash >= 0) {
+								entryName = entryName.replaceFirst(name.substring(0, lastSlash + 1), "");
```

Comment:
Which ever way he does it, as long as it's clear to the reader I'd be okay with it